ledger/catchupaccessor.go

```diff
@@ -289,13 +293,19 @@ func (c *CatchpointCatchupAccessorImpl) processStagingBalances(ctx context.Conte
 	wdb := c.ledger.trackerDB().wdb
 	err = wdb.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
 		// create the merkle trie for the balances
-		mc, err0 := makeMerkleCommitter(tx, true)
-		if err0 != nil {
-			return err0
-		}
-		trie, err := merkletrie.MakeTrie(mc, trieCachedNodesCount)
+		var mc *merkleCommitter
+		mc, err = makeMerkleCommitter(tx, true)
 		if err != nil {
-			return err
+			return
+		}
+
+		if progress.cachedTrie == nil {
+			progress.cachedTrie, err = merkletrie.MakeTrie(mc, trieCachedNodesCount)
```

Comment:
do we hold a lock while assigning?

Reply:
no, why should we ? do you see any potential contention ?
as far as I can tell, only this function is accessing this field.

Reply:
if two threads runs this function then the second thread can reinitialize the cache.

Reply:
which two threads ? there can be only a single catchpoint "restore" at a time.